app/routes/capital-projects.$managingCode.$capitalProjectId.tsx

```diff
@@ -1,3 +1,41 @@
+import { LoaderFunctionArgs, json } from "@remix-run/node";
+import { useLoaderData, useNavigate } from "@remix-run/react";
+import {
+  findCapitalProjectByManagingCodeCapitalProjectId,
+  findAgencies,
+} from "../gen";
+import { CapitalProjectDetailPanel } from "../components/CapitalProjectDetailPanel";
+
+export async function loader({ params }: LoaderFunctionArgs) {
+  const agenciesResponse = await findAgencies({
+    baseURL: `${import.meta.env.VITE_ZONING_API_URL}/api`,
+  });
+  if (
+    typeof params.managingCode === "undefined" ||
+    typeof params.capitalProjectId === "undefined"
+  ) {
```

Comment:
Checking the type is "undefined" strikes me as more verbose and less precise than directly checking the value is literally undefined. Also, we reference each value at least twice. We can de-structure it once and directly use the primitive values; ie)
```suggestion
const { managingCode, capitalProjectId } = params;
  if (
    managingCode === undefined ||
    capitalProjectId === undefined
  ) {
```